clients/client-java/src/main/java/org/apache/gravitino/client/GenericModelCatalog.java

```diff
@@ -0,0 +1,330 @@
+import org.apache.gravitino.exceptions.ModelVersionAliasesAlreadyExistException;
+import org.apache.gravitino.exceptions.NoSuchModelException;
+import org.apache.gravitino.exceptions.NoSuchModelVersionException;
+import org.apache.gravitino.exceptions.NoSuchSchemaException;
+import org.apache.gravitino.model.Model;
+import org.apache.gravitino.model.ModelCatalog;
+import org.apache.gravitino.model.ModelVersion;
+import org.apache.gravitino.rest.RESTUtils;
+
+class GenericModelCatalog extends BaseSchemaCatalog implements ModelCatalog {
+
+  GenericModelCatalog(
+      Namespace namespace,
+      String catalogName,
+      Catalog.Type catalogType,
+      String provider,
+      String comment,
+      Map<String, String> properties,
+      AuditDTO auditDTO,
+      RESTClient restClient) {
+    super(namespace, catalogName, catalogType, provider, comment, properties, auditDTO, restClient);
+  }
+
+  @Override
+  public ModelCatalog asModelCatalog() {
+    return this;
+  }
+
+  @Override
+  public NameIdentifier[] listModels(Namespace namespace) throws NoSuchSchemaException {
+    checkModelNamespace(namespace);
+
+    Namespace modelFullNs = modelFullNamespace(namespace);
+    EntityListResponse resp =
+        restClient.get(
+            formatModelRequestPath(modelFullNs),
+            EntityListResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.modelErrorHandler());
+    resp.validate();
+
+    return Arrays.stream(resp.identifiers())
+        .map(id -> NameIdentifier.of(id.namespace().level(2), id.name()))
+        .toArray(NameIdentifier[]::new);
+  }
+
+  @Override
+  public Model getModel(NameIdentifier ident) throws NoSuchModelException {
+    checkModelNameIdentifier(ident);
+
+    Namespace modelFullNs = modelFullNamespace(ident.namespace());
+    ModelResponse resp =
+        restClient.get(
+            formatModelRequestPath(modelFullNs) + "/" + RESTUtils.encodeString(ident.name()),
+            ModelResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.modelErrorHandler());
+    resp.validate();
+
+    return new GenericModel(resp.getModel());
+  }
+
+  @Override
+  public Model registerModel(NameIdentifier ident, String comment, Map<String, String> properties)
+      throws NoSuchSchemaException, ModelAlreadyExistsException {
+    checkModelNameIdentifier(ident);
+
+    Namespace modelFullNs = modelFullNamespace(ident.namespace());
+    ModelRegisterRequest req = new ModelRegisterRequest(ident.name(), comment, properties);
+    req.validate();
+
+    ModelResponse resp =
+        restClient.post(
+            formatModelRequestPath(modelFullNs),
+            req,
+            ModelResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.modelErrorHandler());
+
+    resp.validate();
+    return new GenericModel(resp.getModel());
+  }
+
+  @Override
+  public boolean deleteModel(NameIdentifier ident) {
```

Comment:
When we have `register...`, I'm expecting a `deregister...` rather than a `delete...`.

Reply:
I refer this from other system's design (https://docs.unitycatalog.io/usage/models/#deleting-registered-models-and-model-versions), I think it is fine and easy to understand.